v2/internal/testcommon/kube_per_test_context.go

```diff
@@ -95,8 +95,15 @@ func (tc KubePerTestContext) MakeReferenceFromResource(resource client.Object) *
 }
 
 func (tc KubePerTestContext) NewTestResourceGroup() *resources.ResourceGroup {
+	return tc.NewTestResourceGroupWithNamespace(tc.Namespace)
+}
+
+func (tc KubePerTestContext) NewTestResourceGroupWithNamespace(namespace string) *resources.ResourceGroup {
```

Comment:
I don't think we really need this: Instead you can just modify the returned `resources.ResourceGroup` before creating it.

```
rg := tc.NewTestResourceGroup()
rg.Namespace = ns1
```

Reply:
Done!